omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -435,60 +426,74 @@ def gen_frontend_default
     end
 
     def gen_api_fqdn_default
-      PrivateChef["lb"]["api_fqdn"] ||= PrivateChef['api_fqdn']
-      PrivateChef["lb"]["web_ui_fqdn"] ||= PrivateChef['api_fqdn']
-      PrivateChef["nginx"]["server_name"] ||= PrivateChef['api_fqdn']
+      PrivateChef["lb"]["api_fqdn"] ||= PrivateChef["api_fqdn"]
+      PrivateChef["lb"]["web_ui_fqdn"] ||= PrivateChef["api_fqdn"]
+      PrivateChef["nginx"]["server_name"] ||= PrivateChef["api_fqdn"]
       PrivateChef["nginx"]["url"] ||= "https://#{PrivateChef['api_fqdn']}"
     end
 
     def gen_secrets_default(node_name)
-      consume_existing_secrets
-      # Transition from erchef's sql_user/password etc living under 'postgresql' in older versions,
-      # to 'opscode_erchef' in newer versions.
-      if PrivateChef['postgresql'].has_key? 'sql_password'
-        PrivateChef['opscode_erchef']['sql_password'] ||= PrivateChef['postgresql']['sql_password']
```

Comment:
This will still be a necessary thing if upgrading from older versions that kept the password in a different place, and I don't see that it's handled below?